### PR TITLE
Feature/data source static view

### PR DESCRIPTION
**Pull request by kalenluciano**
#### Fixes

* Add a new route in the backend to get all info on a specific data source by its ID
* Creates a page to display all info on a specific data source
* Make the data source detail button navigate to the data source static view page
* Separate out the footer and NavBar into their own components
* Add two NavBars (one with just the logo and one with links) that conditionally render depending on the current page
* Change the global.css import to fix a random bug where the CSS would sometimes stop applying to the site

#### Description
The primary purpose of this PR was to add a Data Source Static View page that displays all the available (non-PII) info for a data source. In addition to adding a page, I added a separate route in the backend to get all the necessary info to display. I also altered some formatting and styling, both on the Data Source Static View page and with the NavBars.

### CODE REVIEW

**Review by joshuagraber — CHANGES_REQUESTED** [review]
Looks great! Just a couple suggestions and questions around component naming and error handling.

File: client/src/App.vue [hunk 1]
```diff
@@ -1,21 +1,27 @@
 <template>
-  <img src="@/assets/logo.svg" class="logo"/>
+  <NavBarLinks v-if="showNavLinks()" />
+  <NavBarLogo v-else />
```
[the remark this answers is not in the document]
Comment (kalenluciano) [reply client/src/App.vue:3]: Makes sense to me, added this in

File: client/src/pages/DataSourceStaticView.vue [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        </div>
+        <p class="large">Municipality</p>
+        <p class="small">{{ dataSource.municipality || 'null' }}</p>
+        <p class="large">Agency Type</p>
+        <p class="small">{{ dataSource.agency_type || 'null' }}</p>
+        <p class="large">Jurisdiction Type</p>
+        <p class="small">{{ dataSource.jurisdiction_type || 'null' }}</p>
+      </div>
+
+      <div class="data-detail-section">
```
[the remark this answers is not in the document]
Comment (kalenluciano) [reply client/src/pages/DataSourceStaticView.vue]: Great idea! It definitely felt redundant typing it all out

File: client/src/pages/DataSourceStaticView.vue [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    id: null
+  }),
+  mounted: function() {
+    this.id = this.$route.params.id
+    this.getDataSourceDetails()
+  }, 
+  methods: {
+    async getDataSourceDetails() {
+      const headers = {"Authorization": `Bearer ${process.env.VUE_APP_PDAP_TOKEN}`}
+      const res = await axios.get(`${process.env.VUE_APP_BASE_URL}/data-sources/${this.id}`, {headers})
```
[the remark this answers is not in the document]
Comment (kalenluciano) [reply client/src/pages/DataSourceStaticView.vue:89]: I added some error handling for this

**Review by kalenluciano — COMMENTED** [review]
Thanks for the feedback! I implemented these changes and will push an updated PR shortly.

**Review by joshuagraber — DISMISSED** [review]

**Comment by mbodeantor** [comment]
Are we still not able to run this locally? I'm getting an axios error still

**Comment by kalenluciano** [comment]
I'm getting the same issue. I think I'm going to move these env variables to a globals.js file instead. I'll push updates shortly!

File: client/src/pages/DataSourceStaticView.vue [hunk 4]
```diff
@@ -0,0 +1,229 @@
+<template>
+  <div>
+    <div class="data-details-header">
+      <h2>{{ dataSource.name }}</h2>
+      <button class="button">Edit</button>
```
[the remark this answers is not in the document]
Comment (josh-chamberlain) [reply client/src/pages/DataSourceStaticView.vue:5]: (totally fine if it's removed until the edit function is possible, thanks for sticking to the wireframe)

**Comment by mbodeantor** [comment]
I think you can just remove the hardcoded variable for BASE_URL since it is specified for each env in DigitalOcean

**Comment by mbodeantor** [comment]
Also I don't think "Formats available" is iterating correctly
Screenshot 2023-10-13 at 1 12 01 PM

**Comment by joshuagraber** [comment]
Hey @kalenluciano @mbodeantor I checked out [this commit](https://github.com/Police-Data-Accessibility-Project/data-sources-app/pull/58/commits/82ea4b6367f8f7b7634137f908c765c05426deeb) (the one prior to adding these as js globals), and it's working for me.

I've noticed that I need to re-set the flask env vars every time I start the app, maybe that's it? I added a `zsh` func/alias to activate venv, install python requirements, and set the env stuff. Can send to you in Discord if you're interested.

The Formats available section does need fixing though.

**Comment by kalenluciano** [comment]
I changed the globals.js back to env variables based on our conversation earlier this week. 

Re: > Also I don't think "Formats available" is iterating correctly Screenshot 2023-10-13 at 1 12 01 PM

I made a fix to this in the backend, but since the frontend app is pointing to live deployed version, you won't see this change. To test the change, you'll have to change VUE_APP_BASE_URL to localhost:5000 and run the backend

**Review by mbodeantor — APPROVED** [review]
Awesome, looks good to me!

**Pull request merged**